### PR TITLE
Add method to change ReduceMotion setting for simulators

### PR DESCRIPTION
**Pull request by vrunoa**
fyi @jlipps 

Add the possibility to set ReduceMotion in simulators, this can help reduce flakiness of tests.

### CODE REVIEW

File: lib/settings.js [hunk 1]
```diff
@@ -132,6 +134,17 @@ async function updateLocationSettings (sim, bundleId, authorized) {
   }
 }
 
+async function reduceMotion (sim, reduceMotion = true) {
+  let paths = await plistPaths(sim, 'accesibilitySettings');
+  for (let i in paths) {
```
Comment (mykola-mokhnach) [inline lib/settings.js]: use `of` instead

File: lib/settings.js [hunk 2]
```diff
@@ -132,6 +134,17 @@ async function updateLocationSettings (sim, bundleId, authorized) {
...
+    let file = paths[i];
+    let updates = {
+      ReduceMotionEnabled: reduceMotion ? 1 : 0
```
Comment (mykola-mokhnach) [inline lib/settings.js]: is it numeric or boolean parameter in the actual plist?

File: lib/settings.js [hunk 3]
```diff
@@ -132,6 +134,17 @@ async function updateLocationSettings (sim, bundleId, authorized) {
...
+    };
+    await update(file, updates);
```
Comment (mykola-mokhnach) [inline lib/settings.js]: the intermediate `updates` variable is not needed

File: lib/simulator-xcode-6.js [hunk 4]
```diff
@@ -617,6 +617,25 @@ class SimulatorXcode6 extends EventEmitter {
...
   }
 
+  /**
+   * Enable/Disable reduce motion.
+   *
+   * @param {string} bundleId - The application ID to update.
+   * @param {boolean} reduceMotion - Whether or not to enable it.
+   */
+  async reduceMotion (bundleID, reduceMotion = true) {
+    if (await this.isRunning()) {
```
Comment (mykola-mokhnach) [inline lib/simulator-xcode-6.js]: I would not do daemon killing here, since it will an expected side effect

File: lib/simulator-xcode-6.js [hunk 5]
```diff
@@ -617,6 +617,25 @@ class SimulatorXcode6 extends EventEmitter {
...
+      await this.endSimulatorDaemon();
+    } else if (await this.isFresh()) {
+      await this.launchAndQuit(false, STARTUP_TIMEOUT);
+    }
+    await settings.reduceMotion(this, bundleID, reduceMotion);
+  }
+
+  async readSettings (plist) {
```
Comment (mykola-mokhnach) [inline lib/simulator-xcode-6.js]: what is the purpose of `readSettings` method?

File: test/functional/settings-e2e-specs.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    if (devicePresent) {
+      await killAllSimulators();
+      await simctl.deleteDevice(udid);
+    }
+  });
+
+  it("check accesibility reduce motion settings", async function () {
+    let sim = await getSimulator(udid);
+    await sim.reduceMotion(true);
+    let fileSettings = await sim.readSettings('accesibilitySettings');
```
Comment (mykola-mokhnach) [inline test/functional/settings-e2e-specs.js]: it is, actually, spelled as `accessibility`

File: test/functional/settings-e2e-specs.js [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    for (let file in fileSettings) {
+      let settings = fileSettings[file];
```
Comment (mykola-mokhnach) [inline test/functional/settings-e2e-specs.js]: use `of`

File: lib/settings.js (hunk 2 above)
Comment (vrunoa) [reply lib/settings.js]: @mykola-mokhnach is an integer.

File: lib/simulator-xcode-6.js (hunk 4 above)
Comment (vrunoa) [reply lib/simulator-xcode-6.js]: @mykola-mokhnach the reduce motion change wont be available if we dont reboot the simulator. Let change that kill for reboot.

File: lib/simulator-xcode-6.js (hunk 5 above)
Comment (vrunoa) [reply lib/simulator-xcode-6.js]: I was using it for functional testing, let me see if I can have a different approach besides exposing this method

File: lib/simulator-xcode-6.js [hunk 8]
```diff
@@ -612,6 +612,21 @@ class SimulatorXcode6 extends EventEmitter {
     return await settings.updateLocationSettings(this, bundleId, authorized);
   }
 
+  /**
+   * Enable/Disable reduce motion.
+   *
+   * @param {string} bundleId - The application ID to update.
+   * @param {boolean} reduceMotion - Whether or not to enable it.
+   */
+  async reduceMotion (reduceMotion = true) {
```
Comment (mykola-mokhnach) [inline lib/simulator-xcode-6.js]: I'd rather rename the method to setReduceMotion

File: lib/simulator-xcode-6.js [hunk 9]
```diff
@@ -612,6 +612,21 @@ class SimulatorXcode6 extends EventEmitter {
...
+    if (await this.isRunning()) {
```
Comment (mykola-mokhnach) [inline lib/simulator-xcode-6.js]: I still consider that restart should not be in this method. It should be the caller code, who is responsible for simulator reboot (and it has to be stated in the docstring, that the method has no effect unless the sim is rebooted if it is currently running)

Comment (jlipps) [reply lib/simulator-xcode-6.js]: Yeah, I agree. We have (or used to have) code in appium-ios-driver / appium-xcuitest-driver that check the capabilities a user sends in to determine if any of them will require a sim reboot, and it handles the reboot after running the commands from this package. It should be up to the caller to do the reboot.

File: test/functional/settings-e2e-specs.js [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+      }).length > 0;
+    if (devicePresent) {
+      await killAllSimulators();
+      await simctl.deleteDevice(udid);
+    }
+  });
+
+  it("check accesibility reduce motion settings", async function () {
+    let sim = await getSimulator(udid);
+    await sim.reduceMotion(true);
```
Comment (imurchie) [inline test/functional/settings-e2e-specs.js]: `setReduceMotion`?

Comment (vrunoa) [reply test/functional/settings-e2e-specs.js]: thanks

File: test/unit/settings-specs.js [hunk 11]
```diff
@@ -270,4 +269,31 @@ describe('settings', function () {
...
+      sim = new SimulatorXcode6();
+      sinon.stub(settings, 'setReduceMotion');
+    });
+
+    afterEach(async function () {
+      sinon.restore();
+      sinon.stub(settings, 'setReduceMotion');
+    });
+
+    it('should launch simulator if not fresh before setting reduce motion', async function () {
```
Comment (mykola-mokhnach) [inline test/unit/settings-specs.js]: perhaps `if fresh`

**Pull request merged**